src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,32 @@
 package core.basesyntax.service;
 
+import core.basesyntax.InvalidDataException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_CHARACTERS = 6;
+    private static final int MINIMUM_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidDataException {
+        try {
+            if (user.getAge() == null || user.getLogin() == null || user.getPassword() == null
+                    || user.getId() == null) {
+                throw new InvalidDataException("Ooops, your input data is not valid");
+            }
+        } catch (NullPointerException e) {
+            throw new InvalidDataException("Oops, your input data is not valid");
+        }
+        if (user.getPassword().length() >= MINIMUM_CHARACTERS
+                && user.getLogin().length() >= MINIMUM_CHARACTERS
+                && user.getAge() >= MINIMUM_AGE) {
+            storageDao.add(user);
+            return user;
+        } else {
+            throw new InvalidDataException("Oops, your input data is not valid");
+        }
```

Comment:
this is not readable
* we should split it into methods that will be responsible for checking a single thing, like `validateLogin()` and call them one by one in `register()`
* we should not have any try/catch blocks here, RunTime exceptions like `NullPointerException` should be avoided and not handled by code when we already struck up in ot
* exception messages should be informative, which means it should contain information on what exactly went wrong, was it a null value? or invalid value for the business logic point of view, we also should indicate what value would be valid(putting min age to that message for example)